pkgs/tools/admin/yandexcloud-cli/default.nix

```diff
@@ -0,0 +1,104 @@
+{ lib
+, stdenv
+, fetchurl
+, makeBinaryWrapper
+, installShellFiles
+, testers
+  # update script
+, writeShellScript
+, curl
+, common-updater-scripts
+}:
+let
+  version = "0.109.0";
+  hashes = {
+    "aarch64-darwin" = "sha256-Fyt41VyjWuaRlF4ulwSo2yTX9jlBh/+syv53UxrC+jE=";
+    "aarch64-linux" = "sha256-Dy3jxSxTBbuHJEKcchVlayIlUANkPMxuYcksO+18Te0=";
+    "i686-linux" = "sha256-yG2ZSdZ0c+zf3cAn4Kkomlck5EjS8I0MHcJndInW4Ik=";
+    "x86_64-darwin" = "sha256-iRB1+aYsXptR4ksIa0vtWIucx5aiqGy676fBDuvN/e4=";
+    "x86_64-linux" = "sha256-2Vul/LrcdGncwf571Px06AwBKAt/qmlIUvbWC6TtWkI=";
+  };
+
+  storageURL = "https://storage.yandexcloud.net/yandexcloud-yc/release";
+
+  fetchSource = system: hash:
+    let
+      throwSystem = throw "Unsupported system: ${system}";
+      platform = lib.systems.elaborate system;
+      arch =
+        if platform.isAarch64 then "arm64"
+        else if platform.isx86_64 then "amd64"
+        else if platform.isx86_32 then "386"
+        else throwSystem;
+      os =
+        if platform.isLinux then "linux"
+        else if platform.isDarwin then "darwin"
+        else throwSystem;
+    in
+    fetchurl {
+      url = "${storageURL}/${version}/${os}/${arch}/yc";
+      inherit hash;
+      executable = true;
+    };
+
+  sources = lib.mapAttrs fetchSource hashes;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "yandexcloud-cli";
+  inherit version;
+
+  src = sources.${stdenv.hostPlatform.system};
+
+  dontUnpack = true;
+
+  strictDeps = true;
+
+  nativeBuildInputs = [ installShellFiles makeBinaryWrapper ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp -a $src $out/bin/yc
+  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    installShellCompletion --cmd yc \
+      --bash <($out/bin/yc completion bash) \
+      --zsh <($out/bin/yc completion zsh)
+  '' + ''
+    makeWrapper $out/bin/yc $out/bin/docker-credential-yc \
+      --add-flags --no-user-output \
+      --add-flags container \
+      --add-flags docker-credential
+
+    runHook postInstall
+  '';
+
+  passthru = {
+    inherit sources;
```

Comment:
Any reason why?

Reply:
So that `update-source-version` can locate the expression to update hashes for each platform.
See [`rg -- --source-key=\"sources`](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/NixOS/nixpkgs%24+--source-key%3D%22sources) for other (but likely not all) examples of this pattern.

Reply:
Understood.